src/System.Dynamic.Runtime/src/System/Linq/Expressions/DynamicExpression.cs

```diff
@@ -1214,9 +1214,14 @@ private static DynamicExpression MakeDynamic(CallSiteBinder binder, Type returnT
             }
         }
 
-        private static void ValidateDynamicArgument(Expression arg)
+        private static void ValidateDynamicArgument(Expression arg, string paramName)
         {
-            ExpressionUtils.RequiresCanRead(arg, "arguments");
```

Comment:
This was a bug; for N-ary overloads, the `paramName` reported was the non-existent `arguments`.